File: isPrime.py
import math

while True:
	var = input("Enter a number: ")
	number = int(var)
	if number < 2:
		break
	divNumber = 2
	isDiv = False

	if number > 2:
		for x in range(2, math.ceil(math.sqrt(number)) + 1):
			if number % x == 0:
				isDiv = True
				divNumber = x
				break

	if isDiv:
		print("Number is NOT prime. Divisible by %d" % (divNumber))
	else:
		print("Number is Prime")

File: isPrime.cpp
#include <iostream>
#include <cmath>

using namespace std;

void checkPrime(int num) {
    int divNum = 2;
    bool isDiv = false;
    if(num > 2) {
        for (int x = 2; x < ceil(sqrt(num)) + 1; x++) {
	    if(num % x == 0) {
	        isDiv = true;
                divNum = x;
		break;
            }
        }
    }
    if(isDiv) {
        cout << "Number is NOT prime. Divisible by " << divNum << endl;
    } else {
        cout << "Number is Prime." << endl;
    }
}


int main()
{
    int num;
    while(true) {
        cout << "Enter a number: ";
        cin >> num;
        if(num < 2) {
            break;
        }
        checkPrime(num);
    }
    return 0;
}
